### PR TITLE
Conda build set-up

### PR DESCRIPTION
**Pull request by SgStapleton**
This begins to address https://github.com/USGS-Astrogeology/ale/issues/4

### CODE REVIEW

**Comment by SgStapleton** [comment]
The ``conda build``` command appears to run successfully, but I have not tested pushing this to anaconda, or (obviously) installing it from the usgs-astrogeology channel.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  git_url: 'https://github.com/USGS-Astrogeology/ale.git'
+  git_branch: {{ git_branch }}
+
+build:
+  number: {{ build_number }}
+
+# Shotgun strat on requirements until we can narrow them down
+requirements:
+  build:
+  - python=3.7.1
```
Comment (jlaura) [inline recipe/meta.yaml]: Can we unpin the python version?

**Comment by SgStapleton** [comment]
Waiting for permission to push new packages to anaconda for testing purposes.

File: recipe/meta.yaml (hunk 1 above)
Comment (SgStapleton) [reply recipe/meta.yaml]: It appears to be building properly without this pinned. Will confirm when I can test the uploading and installing process.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  version: {{ version }}
+
+source:
+  git_url: 'https://github.com/USGS-Astrogeology/ale.git'
+  git_branch: {{ git_branch }}
+
+build:
+  number: {{ build_number }}
+
+# Shotgun strat on requirements until we can narrow them down
```
Comment (kberryUSGS) [inline recipe/meta.yaml]: I think comment this is from the ISIS3 one and not applicable here?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  - python-dateutil
+  - pvl
+
+# Add the tests eventually
+# test:
+#
+
+# about:
+#   home: https://developers.google.com/protocol-buffers/
+#   license: BSD 3-Clause
```
Comment (kberryUSGS) [inline recipe/meta.yaml]: What license do we want to use for ale?

Comment (SgStapleton) [reply recipe/meta.yaml]: Yeah, I left an example "about" section commented out for reference, but forgot to ask about specific ale implementation... Any thoughts?

Comment (kberryUSGS) [reply recipe/meta.yaml]: Yeah! This is less a comment on your actual PR and more a question that should have gone in a ticket. I'm not sure.

File: recipe/build.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=$PREFIX ..
```
Comment (jessemapel) [inline recipe/build.sh]: We should add a `-DCMAKE_BUILD_TYPE=Release` here

File: recipe/meta.yaml (hunk 3 above)
Comment (jessemapel) [reply recipe/meta.yaml]: I think we can't use a license? My understandings is that we have to push everything we do into the public domain. @jlaura has looked into this more.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+#   license_file: LICENSE
+#   summary: "Protocol Buffers - Google's data interchange format."
```
Comment (jessemapel) [inline recipe/meta.yaml]: Probably should change this summary

File: recipe/meta.yaml (hunk 3 above)
Comment (kberryUSGS) [reply recipe/meta.yaml]: @jessemapel Isn't pushing something into the public domain still the sort of thing that requires a license to indicate that it is all in the public domain?

File: recipe/meta.yaml [hunk 6]
```diff
@@ -13,37 +13,54 @@ source:
...
   number: {{ build_number }}
 
+# A number of the requirements below will have two instances: one will include a specific version
+# pinning, while the other one will not. When building a version of ALE for use specifically within
+# ISIS, use the instances with the version pinned. For all other builds, use the unpinned instances.
+# (Confirm that every requirement has one and only one instance uncommented.)
+#
+# ISIS requires pinned versions for these dependencies to avoid conflicts with other dependencies.
+# Make sure to always confirm that version numbers in this file and version numbers in the
+# meta.yaml file within the ISIS3 repo match for each edpendency before building this library
```
Comment (kberryUSGS) [inline recipe/meta.yaml]: typo checking: edependency -> dependency

**Comment by kberryUSGS** [comment]
@SgStapleton Is this ready to be merged?

**Pull request merged**